internal/utils/cli/error.go

```diff
@@ -0,0 +1,138 @@
+package cli
+
+import (
+	"fmt"
+)
+
+// ErrUsageHider may hide the command's usage when an error occurs
+type ErrUsageHider interface {
+	HideUsage() bool
+}
+
+// ErrSuggester provides a list of suggestions that will display to the user when an error occurs
+type ErrSuggester interface {
```

Comment:
you are right that these things don't necessarily _need_ to be specific to errors, but that's currently the only way they can and are used for the CLI currently.

a big part of the command pattern in this CLI is that the command should be self-contained.  if there's output that is relevant to the success path of the command that ran, then the `Handler` func should own that.  it prevents us from having to traverse many code paths in determining how/why a command outputs the things it does.

to further enforce this pattern, the `Handler` interface itself was designed to only return `error` (and no other value)

when it came to the `cobra` framework and how they handled errors, it became clear there were many paths an error could surface and one place to make sure we handled _all_ of them: the return value of `cmd.Execute`.  because of that, we were forced to somehow try and be clever with the error handling.  as you can see from the previous versions of this code, that fell short.